### PR TITLE
Code Review: Dialogs opened by plugins should not show the Sketch icon (#17039)

### PR DESCRIPTION
**Pull request by mathieudutour**
Resolves BohemianCoding/Sketch#17039

### CODE REVIEW

File: Source/ui/UI.js [hunk 1]
```diff
@@ -33,8 +33,13 @@ export function message(text, document) {
  * by pressing the OK button.
  */
 export function alert(title, text) {
-  const app = NSApplication.sharedApplication()
-  app.displayDialog_withTitle(text, title)
+  const dialog = NSAlert.alloc().init()
+  dialog.setMessageText(title)
+  dialog.setInformativeText(text)
+  if (__command.pluginBundle() && __command.pluginBundle().icon()) {
+    dialog.icon = __command.pluginBundle().icon()
```
Comment (opsGavin) [inline Source/ui/UI.js]: If the plugin hasn't supplied it's own icon would this still default to the sketch icon? Wondering if we need an `else` to set the plugin icon here instead?

File: Source/ui/UI.js [hunk 2]
```diff
@@ -33,8 +33,13 @@ export function message(text, document) {
  * by pressing the OK button.
  */
 export function alert(title, text) {
-  const app = NSApplication.sharedApplication()
-  app.displayDialog_withTitle(text, title)
+  const dialog = NSAlert.alloc().init()
```
Comment (opsGavin) [inline Source/ui/UI.js]: BTW - in Obj-C you can do `[class new]` as a shortcut to `[[class alloc] init]`. It doesn't matter, but I expect you could do `NSAlert.new()` here.

File: Source/ui/UI.js (hunk 1 above)
Comment (mathieudutour) [reply Source/ui/UI.js]: Ah yes that'd be great indeed, good call

File: Source/ui/UI.js (hunk 2 above)
Comment (mathieudutour) [reply Source/ui/UI.js]: indeed but I find it confusing with the `new` keyword in javascript :)

Comment (opsGavin) [reply Source/ui/UI.js]: Fair enough :)

**Pull request merged**